unittests/BuildSystem/BuildSystemTaskTests.cpp

```diff
@@ -181,4 +184,62 @@ TEST(BuildSystemTaskTests, directorySignature) {
   ASSERT_TRUE(resultA->toData() != resultC->toData());
 }
 
+TEST(BuildSystemTaskTests, doesNotProcessDependenciesAfterCancellation) {
+  TmpDir tempDir{ __FUNCTION__ };
+
+  std::string outputFile = tempDir.str() + "/output.txt";
+
+  SmallString<256> manifest{ tempDir.str() };
+  sys::path::append(manifest, "manifest.llbuild");
```

Comment:
The only reason you had to define this inline was because we can't currently create the ShellCommand from here, right?

I think we should probably expose those types so that we can unit test more effectively, what do you think?